src/main/java/org/maxgamer/quickshop/watcher/UpdateWatcher.java

```diff
@@ -147,19 +126,18 @@ public static void uninit() {
 
     @EventHandler
     public void playerJoin(PlayerJoinEvent e) {
-        if (!hasNewUpdate
+        if (this.updater.isLatest(this.updater.getCurrentRunning())
```

Comment:
It will invoke a web request which blocking server thread, we should put the update checking to a async task and just check a boolean value provide by updater.

Reply:
same above